Problem opening image by command line

**Describe the bug**

When I try to open a raw file by `darktable filename` I get the error `error loading: [filename]`. However when I do it again, the image loads correctly. It doesn't seem to matter if an instance of darktable was already running or not. Moreover sometimes I get duplicate "film rolls" for the same folder in lighttable view with some pictures in one and some in the other. I'm attaching parts of `darktable -d all` output for the described actions. I'm running current HEAD. It seems that the bug was introduced fairly recently as I recompile Darktable quite often and I noticed it only today. I can try bisect if it would help.

I tried the same after removing `.config/darktable`. First image seemed to open fine, but second or so produced the error. I also noticed once that newly opened image had a wrong thumbnail in darktable "film strip" area - thumbnail from completely different picture which I processed a month ago.

**To Reproduce**
1. Run `darktable filename` once or a few times for different files
2. See error

**Platform (please complete the following information):**
 - Darktable Version: HEAD 8e4d22171fafc9546c6e775b03eaa5c3e0b099b1
 - OS: Gentoo Linux
 - OpenCL: yes or no, no change
 - AMD Radeon Fury

Hmm, I don't know why it didn't work. Maybe something got messed up with ninja or cmake, or maybe I only pulled and forgot to build. Anyway I just tried clean build (before I was just rebuilding existing tree) and I got the compilation failure in some other place :/ 
```
../src/dtgtk/thumbnail.c: In function ‘_thumb_update_icons’:
../src/dtgtk/thumbnail.c:408:5: error: unknown type name ‘dt_variables_params_t’
  408 |     dt_variables_params_t *vp;
      |     ^~~~~~~~~~~~~~~~~~~~~
../src/dtgtk/thumbnail.c:409:5: error: implicit declaration of function ‘dt_variables_params_init’ [-Werror=implicit-function-declaration]
  409 |     dt_variables_params_init(&vp);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
../src/dtgtk/thumbnail.c:411:7: error: request for member ‘filename’ in something not a structure or union
  411 |     vp->filename = input_dir;
      |       ^~
../src/dtgtk/thumbnail.c:412:7: error: request for member ‘jobcode’ in something not a structure or union
  412 |     vp->jobcode = "infos";
      |       ^~
../src/dtgtk/thumbnail.c:413:7: error: request for member ‘imgid’ in something not a structure or union
  413 |     vp->imgid = thumb->imgid;
      |       ^~
../src/dtgtk/thumbnail.c:414:7: error: request for member ‘sequence’ in something not a structure or union
  414 |     vp->sequence = 0;
      |       ^~
../src/dtgtk/thumbnail.c:416:18: error: implicit declaration of function ‘dt_variables_expand’ [-Werror=implicit-function-declaration]
  416 |     gchar *msg = dt_variables_expand(vp, pattern, TRUE);
      |                  ^~~~~~~~~~~~~~~~~~~
../src/dtgtk/thumbnail.c:416:18: error: initialization of ‘gchar *’ {aka ‘char *’} from ‘int’ makes pointer from integer without a cast [-Werror=int-conversion]
../src/dtgtk/thumbnail.c:418:5: error: implicit declaration of function ‘dt_variables_params_destroy’ [-Werror=implicit-function-declaration]
  418 |     dt_variables_params_destroy(vp);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

I'm using gcc-9.3.0.

I always do clean builds.

Make sure you've got clean tree and rebuild. that compilation error is from
lighttable rebuild


wt., 24 mar 2020 o 21:36 Tomasz Goliński <notifications@github.com>
napisał(a):

> Hmm, I don't know why it didn't work. Maybe something got messed up with
> ninja or cmake, or maybe I only pulled and forgot to build. Anyway I just
> tried clean build (before I was just rebuilding existing tree) and I got
> the compilation failure in some other place :/
>
> ../src/dtgtk/thumbnail.c: In function ‘_thumb_update_icons’:
>
> ../src/dtgtk/thumbnail.c:408:5: error: unknown type name ‘dt_variables_params_t’
>
>   408 |     dt_variables_params_t *vp;
>
>       |     ^~~~~~~~~~~~~~~~~~~~~
>
> ../src/dtgtk/thumbnail.c:409:5: error: implicit declaration of function ‘dt_variables_params_init’ [-Werror=implicit-function-declaration]
>
>   409 |     dt_variables_params_init(&vp);
>
>       |     ^~~~~~~~~~~~~~~~~~~~~~~~
>
> ../src/dtgtk/thumbnail.c:411:7: error: request for member ‘filename’ in something not a structure or union
>
>   411 |     vp->filename = input_dir;
>
>       |       ^~
>
> ../src/dtgtk/thumbnail.c:412:7: error: request for member ‘jobcode’ in something not a structure or union
>
>   412 |     vp->jobcode = "infos";
>
>       |       ^~
>
> ../src/dtgtk/thumbnail.c:413:7: error: request for member ‘imgid’ in something not a structure or union
>
>   413 |     vp->imgid = thumb->imgid;
>
>       |       ^~
>
> ../src/dtgtk/thumbnail.c:414:7: error: request for member ‘sequence’ in something not a structure or union
>
>   414 |     vp->sequence = 0;
>
>       |       ^~
>
> ../src/dtgtk/thumbnail.c:416:18: error: implicit declaration of function ‘dt_variables_expand’ [-Werror=implicit-function-declaration]
>
>   416 |     gchar *msg = dt_variables_expand(vp, pattern, TRUE);
>
>       |                  ^~~~~~~~~~~~~~~~~~~
>
> ../src/dtgtk/thumbnail.c:416:18: error: initialization of ‘gchar *’ {aka ‘char *’} from ‘int’ makes pointer from integer without a cast [-Werror=int-conversion]
>
> ../src/dtgtk/thumbnail.c:418:5: error: implicit declaration of function ‘dt_variables_params_destroy’ [-Werror=implicit-function-declaration]
>
>   418 |     dt_variables_params_destroy(vp);
>
>       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
>
> cc1: all warnings being treated as errors
>
>
> I'm using gcc-9.3.0.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/darktable-org/darktable/issues/4544#issuecomment-603491074>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACRRKFLCD3CSS4TWK2DGGO3RJEKTJANCNFSM4LS3VM2A>
> .
>


-- 
Pozdrawiam,
Hubert Kowalski

--- a/src/dtgtk/thumbnail.c
+++ b/src/dtgtk/thumbnail.c
@@ -26,6 +26,7 @@
 #include "common/image_cache.h"
 #include "common/ratings.h"
 #include "common/selection.h"
+#include "common/variables.h"
 #include "control/control.h"
 #include "dtgtk/button.h"
 #include "dtgtk/icon.h"

